### PR TITLE
Butler/#143

### PR DESCRIPTION
**Pull request by ButlerATTS**
The changes in this PR fix the bug described in this ticket: https://app.dework.xyz/alchemix/alchemix-v2-frontend/overview?taskId=af1e6eb3-8bac-488b-859a-82a0ac15f5ba

Specifically these changes:
(1) Refreshes user balances in when the users switch wallets
(2) Reloads the page when the user is switching networks within their wallet (for example changing from Ethereum to Optimism)

### CODE REVIEW

File: src/components/composed/StateManager.svelte [hunk 1]
```diff
@@ -29,6 +29,11 @@
 
   async function initialize(netId) {
     if (!initStarted && $addressStore !== undefined && netId !== lastConnection.chainId) {
+      // Reload the page if the vaults were previously loaded and the user is switching chains
+      if (!$vaultsLoading && netId !== lastConnection.chainId) {
+        location.reload();
```
Comment (n4n0GH) [inline src/components/composed/StateManager.svelte]: This is just doing a blatant refresh of the entire page which leads to a new batch or API requests being sent to the respective endpoints. We need a more elegant solution that doesn't create additional queries to keep within API request limits.

**Review by n4n0GH — CHANGES_REQUESTED** [review]

File: src/components/composed/StateManager.svelte (hunk 1 above)
Comment (ButlerATTS) [reply src/components/composed/StateManager.svelte]: I've removed this solution and fixed it by more thoroughly by removing the undefined reference errors that were causing this issue. Please see the latest commit for more details.

File: src/stores/v2/derived.ts [hunk 2]
```diff
@@ -76,15 +76,21 @@ export const vaultsAggregatedCoveredDebt = derived([vaultsStore, balancesStore],
...
-        const underlyingTokenData = getTokenDataFromBalances(_currentVault.underlyingAddress, [$balances]);
-        const normalizedAmount = _currentVault.balance
-          .mul(_currentVault.underlyingPerShare)
-          .div(BigNumber.from(10).pow(underlyingTokenData.decimals))
-          .mul(BigNumber.from(10).pow(BigNumber.from(18).sub(underlyingTokenData.decimals)));
-
-        return _prevVault.add(normalizedAmount);
-      }, initialVal),
+      [Number(vaultTypeKey)]:
+        (vaultBody)
```
Comment (n4n0GH) [inline src/stores/v2/derived.ts:80]: I'm actually unsure if this check is necessary; line 68 already checks for `$vaults` so the const `{vaultBody}` should never be undefined.

**Review by n4n0GH — APPROVED** [review]

File: src/stores/v2/derived.ts (hunk 2 above)
Comment (ButlerATTS) [reply src/stores/v2/derived.ts:80]: Could've sworn I saw this throwing an error at some point, but it looks like you're right. I reverted the code and ran tests on all button/toggle combinations for vault items and saw no issues. I've made a new push that reverts this to my branch, but I'm not sure if it's worth it to you for me to make a new PR for such a small change. Let me know

**Pull request merged**